activesupport/lib/active_support/cache/coder.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+require_relative "entry"
+
+module ActiveSupport
+  module Cache
+    class Coder # :nodoc:
+      def initialize(serializer, compressor, legacy_serializer: false)
+        @serializer = serializer
+        @compressor = compressor
+        @legacy_serializer = legacy_serializer
+      end
+
+      def dump(entry)
+        return @serializer.dump(entry) if @legacy_serializer
+
+        dump_compressed(entry, Float::INFINITY)
+      end
+
+      def dump_compressed(entry, threshold)
+        return @serializer.dump_compressed(entry, threshold) if @legacy_serializer
+
+        # If value is a string with a supported encoding, use it as the payload
+        # instead of passing it through the serializer.
+        if type = type_for_string(entry.value)
+          payload = binary_compatible_string(entry.value)
+        else
+          type = OBJECT_DUMP_TYPE
+          payload = serialize(entry.value)
+        end
+
+        # Compress payload if it meets the size threshold. (Note that `type < 0`
+        # signifies a compressed payload with `type.abs` as its actual type.)
+        if compressed = try_compress(payload, threshold)
+          payload = compressed
+          type = -type
+        end
+
+        expires_at = entry.expires_at || -1.0
+
+        version = dump_version(entry.version)
+        version_length = version&.bytesize || -1
```

Comment:
```suggestion
        if entry.version
          version = dump_version(entry.version)
          version_length = version.bytesize
        else
          version = nil
          version_length = -1
        end
```

I think a little bit more explicitness would be preferable here, this also allow to remove the nil check in `dump_version`.

Reply:
Since this method is a bit long, how do you feel about a middle ground?

```ruby
version = dump_version(entry.version) if entry.version
version_length = version&.bytesize || -1
```

Reply:
Ok 👍